File: app/src/main/java/com/abhishek/remotetest/RemoteTestApp.java
package com.abhishek.remotetest;

import android.app.Application;

import com.abhishek.remotetest.domain.StoreDatabase;

public class RemoteTestApp extends Application {

    private static StoreDatabase appDatabase;

    @Override
    public void onCreate() {
        super.onCreate();

        initializeDb();
    }

    private void initializeDb() {
        appDatabase = StoreDatabase.Companion.getInstance(getApplicationContext());
    }

    public static StoreDatabase getAppDatabase() {
        return appDatabase;
    }
}

File: app/src/main/java/com/abhishek/remotetest/RemoteTestApp.kt
package com.abhishek.remotetest

import android.app.Application
import com.abhishek.remotetest.domain.StoreDatabase

public class RemoteTestApp : Application() {
    private var appDatabase: StoreDatabase? = null

    override fun onCreate() {
        super.onCreate()
        initializeDb()
    }

    private fun initializeDb() {
        appDatabase = StoreDatabase.Companion.getInstance(applicationContext)
    }

    fun getAppDatabase(): StoreDatabase? {
        return appDatabase
    }
}
